cron/cron.go

```diff
@@ -271,18 +270,25 @@ func (c *Cron) run() {
 		// Determine the next entry to run.
 		sort.Sort(byTime(c.entries))
 
-		var timer *clock.Timer
+		var (
+			timer   clock.Timer
+			timerCh <-chan time.Time
+		)
 		if len(c.entries) == 0 || c.entries[0].Next.IsZero() {
 			// If there are no entries yet, just sleep - it still handles new entries
 			// and stop requests.
-			timer = c.clk.Timer(100000 * time.Hour)
+			timerCh = make(chan time.Time)
 		} else {
-			timer = c.clk.Timer(c.entries[0].Next.Sub(now))
+			timer = c.clk.NewTimer(c.entries[0].Next.Sub(now))
+			timerCh = timer.C()
 		}
 
 		for {
 			select {
-			case now = <-timer.C:
+			case now = <-timerCh:
+				// Set timer to nil so we can exit cleanly
+				timer = nil
```

Comment:
why are we setting timer to nil here? 
Will the clock ticks still work if the timer reference is gone? Won't it be garbage collected?

Reply:
If this isn’t set to nil, below on L317 we would try to call Stop() again on the timer which has already fired. This will cause a deadlock. (It’s the reason why we had https://github.com/dapr/dapr/pull/6523